### ISSUE TITLE
Logistic regression 

### ISSUE THREAD

**Issue by freeloophulk** [ysebastian-student-work-cdu/PRT564-GROUP-11-DATA-ANALYTICS#2]
# import models 
import pandas as pd
import numpy as np
from sklearn.datasets import make_classification
from matplotlib import pyplot as plt
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score 
from sklearn.metrics import confusion_matrix
import seaborn as sns 
import statsmodels.api as sm

df = pd.read_csv("databreaches_tra.csv")

fig = plt.figure(figsize=(25,20))

sns.heatmap(df.corr(), annot=True)
plt.savefig("Heatmap.png")

x = df.iloc[:,11:].values
y = df.iloc[:,6].values

# split the train and test sets
x_train, x_test, y_train, y_test = train_test_split(x,y, test_size=0.4, random_state= 0)

log_reg = LogisticRegression()

# Train the model 
log_reg.fit(x_train,y_train)

# make predictions 
 
y_pre = log_reg.predict(x_test)
y_act = y_test

from sklearn.metrics import classification_report
print(classification_report(y_test, y_pre))

#logit_model=sm.Logit(x,y)

#result=logit_model.fit()
#print(result.summary2())

#measure accuracy 
[databreaches_tra.csv](https://github.com/ysebastian-student-work-cdu/PRT564-GROUP-11-DATA-ANALYTICS/files/11331489/databreaches_tra.csv)

print ('train accuracy:',np.round(accuracy_score(y_train,log_reg.predict(x_train)),2))
print ('test accuracy:',np.round(accuracy_score(y_act,y_pre),2))
# make the confusion matrix 
cf = confusion_matrix(y_test,y_pre)
print (cf)
print("\nTest confusion_matrix")
sns.heatmap(cf, annot=True, cmap='Blues')
plt.xlabel('Predicted', fontsize=12)
plt.ylabel('Actural',fontsize=12)
plt.savefig("confusion_matrix.png")
